examples/import_catalog_items/script.py

```diff
@@ -0,0 +1,223 @@
+    workflow creating multiple catalog items.
+    """
+    def __init__(self, client: push_to_3yourmind.PushTo3YourmindAPI):
+        self.client = client
+        
+    def list_form_details(self):
+        return self.client.common.get_forms()
+        
+    def create_basket(self):
+        return self.client.user_panel.create_basket()
+
+    def configure_basket_line(
+            self,
+            *,
+            basket_id: int,
+            file: t.Union[str, io.BytesIO],
+            product_id: int,
+            post_processings: t.List[push_to_3yourmind.td.PostProcessingConfig] = (),
+            part_requirements: t.Optional[push_to_3yourmind.td.FormData] = None,
+    ):
+        line = self.client.user_panel.create_line_with_cad_file_and_product(
+            basket_id=basket_id,
+            cad_file=file,
+            quantity=1,
+            product_id=product_id,
+            post_processings=post_processings,
+        )
+        if part_requirements is not None:
+            self.client.user_panel.add_part_requirements_to_basket_line(
+                line_id=line["id"],
+                form_data=part_requirements,
+            )
+        return line
+    
+    def create_catalog_item(self, *, basket_line_id: int):
+        return self.client.user_panel.create_catalog_item(
+            basket_line_id=basket_line_id,
+        )
+
+    def add_attachments_to_catalog_item(
+            self,
+            catalog_item_id: int,
+            attachment: t.Union[str, io.BytesIO],
+    ):
+        self.client.user_panel.upload_catalog_item_attachment(
+            catalog_item_id=catalog_item_id,
+            attachment_file=attachment,
+        )
+
+    def delete_basket(self, basket_id: int):
+        # Clean up helper basket
+        self.client.user_panel.delete_basket(basket_id=basket_id)
+        
+
+def import_catalog_items(conf: Config, mapper: MapperFunction, is_dry_run=True) -> None:
+    """
+    Will do all transformations and api calls to turn
+
+    :param conf: Contains base config for this process
+    :param mapper: function that will transform a line in a csv file
+                   and turns it into an ItemData struct
+    """
+    client = push_to_3yourmind.PushTo3YourmindAPI(
+        access_token=conf.ACCESS_TOKEN,
+        base_url=conf.BASE_URL,
+    )
+
+    api_client = ImportCatalogClient(client=client)
+
+    basket = api_client.create_basket()
+
+    logger.info("Start to read csv files")
+    with open(conf.CSV_PATH) as f:
+        csv_reader = csv.DictReader(f)
+
+        for index, line in enumerate(csv_reader):
+            logger.info(f"Reading line {index} of csv file")
+            try:
+                item_data = mapper(line)
+            except MapperException as e:
+                logger.error(f"Mapping Failed: {e}")
+                continue
+
+            try:
+                basket_line = api_client.configure_basket_line(
```

Comment:
@HansAuger I kind of expected to have this logic also more abstracted away inside the library itself.

(1) I thought we could have a function like `api_client.create_catalog_item(file, product_id, part_requirements, post_processings, attachments)` as part of the library. So that the exact order API calls are hidden from the caller...


What do you think?

Reply:
Aye. To explain a little bit, the idea I had was to make the process somewhat more friendly, for the time when the mapping logic is not fully there yet, so you may want to do some testing rounds where you can adjust the code and verify that the part requirements, materials, etc get mapped correctly. For that process it would be annoying if you create incomplete catalog items, that have to be removed later. So that's when I thought it would make sense to disentangle a bit. 

But I see your point and it makes more sense to have one big method with `is_dry_run` as a kwarg